src/test/java/harness/TestCommand.java

```diff
@@ -101,14 +101,23 @@ public static void runCommandExpectSuccess(String... args) {
     assertEquals(0, cmd.exitCode, "exit code = success");
   }
 
+  /**
+   * Helper method to run a command, check its exit code is 0=success, and return {@link Result}.
```

Comment:
This test code took me a while to understand:

```
    JSONObject resolved =
        new JSONObject(
            TestCommand.runAndGetResultExpectSuccess("resource", "resolve", "--name=" + name)
                .stdOut);
```

Instead could we have:

```
    JSONObject resolved =
            TestCommand.runAndGetJSONObjectExpectSuccess("resource", "resolve", "--name=" + name);
```

Ie put `new JSONObject` and `.stdOut` in this helper method?

Reply:
I thought about having a method to return JSONObject specifically but i like the flexibility of returning `Result` because later this will be a jsonarray of jsonobject instead of a jsonobject for datasource resolve. but since this run command have to succeed, i only need to return stdOut. I will change to that and return string.

Reply:
i don't follow. both JsonObject and JsonArray are JsonValue but I would still need to create either one of them in the helper method. How would i know which one to create?

Reply:
I was thinking, have the helper method return JsonValue. In the test, you can cast/convert to JsonObject or JsonArray.

Now I'm thinking, can you have two helper methods, one that returns JsonObject and one that returns JsonArray? I think it's worth it to make the tests easier to read.

```
Use this when there's no Java class (eg UDFGitRepo.class) for the returned JsonObject
runAndGetJsonObjectExpectSuccess

Use this when there's no Java class for the returned JsonArray
runAndGetJsonArrayExpectSuccess
```

Reply:
jsonvalue doesn't have a constructor.

i don't think having new JSONObject(string) is hard to read but i can do it this way if you like.